apollo-runtime/src/main/java/com/apollographql/apollo/ApolloCall.java

```diff
@@ -40,19 +98,40 @@ public void onHttpError(@Nonnull ApolloHttpException e) {
       }
     }
 
+    /**
+     * Gets called when an http request error takes place due to network failures, timeouts etc.
+     */
     public void onNetworkError(@Nonnull ApolloNetworkException e) {
       onFailure(e);
     }
 
+    /**
+     * Gets called when the network request succeeds but there was an error parsing the response.
+     */
     public void onParseError(@Nonnull ApolloParseException e) {
       onFailure(e);
     }
   }
 
+  /**
+   * Factory for creating ApolloCall & ApolloPrefetch objects.
```

Comment:
The purpose of this class is to create Call & Prefetch objects. So felt that factory can be used, though if you feel it is unsuitable, we can change it to something like `creates ApolloCall & ApolloPrefetch`.